app/models/legal_aid_application.rb

```diff
@@ -44,6 +43,13 @@ class LegalAidApplication < ApplicationRecord
     _prefix: true
   )
 
+  def bank_transactions
+    set_transaction_period unless transaction_period_start_at? && transaction_period_finish_at?
+    applicant.bank_transactions.where(
+      happened_at: transaction_period_start_at..transaction_period_finish_at
```

Comment:
I think this filtering could be left out. And we could trust TrueLayer that they returned us the period we asked

Reply:
You're assuming the applicant won't have multiple applications. I thought the filtering could be added at minimal cost, and would prevent inconsistencies.